integration/findandmodify_test.go

```diff
@@ -122,6 +122,32 @@ func TestFindAndModifyErrors(t *testing.T) {
 			},
 			altMessage: "BSON field 'upsert' is the wrong type 'string', expected types '[bool, long, int, decimal, double]'",
 		},
+		"DollarPrefixedFieldName": {
```

Comment:
I think if we can use compat tests, we should use them.
Why don't these cases use compat tests?

Reply:
Yes, these ones can be in compat!